lib/src/configuration.cc

```diff
@@ -841,8 +835,15 @@ void Configuration::validateAndNormalizeWebsocketSettings()
         }
     }
 
-    for (auto const& uri : master_uris_) {
-        validate_https(uri, "master-uris");
+    for (auto &uri : master_uris_) {
+        auto parsed = lth_util::uri(uri);
+        if (parsed.protocol.empty()) {
+            parsed.protocol = "https";
```

Comment:
should we reject http:// ?

Reply:
Hmm, I guess curl is going to explicitly use HTTPS. I'll restore throwing if we specify something besides HTTPS.